graphistry/umap_utils.py

```diff
@@ -126,17 +125,17 @@ def reuse_umap(g: Plottable, memoize: bool, metadata: Any):  # noqa: C901
     )
 
 
-def umap_graph_to_weighted_edges(umap_graph, engine, cfg=config):
+def umap_graph_to_weighted_edges(umap_graph, engine, knn, cfg=config):
     logger.debug("Calculating weighted adjacency (edge) DataFrame")
     coo = umap_graph.tocoo()
     src, dst, weight_col = cfg.SRC, cfg.DST, cfg.WEIGHT
-    if engine == "cuml":
+    if (knn is not None) or (engine == "umap_learn"):
```

Comment:
I'm confused...  If someone passes knn with engine=cuml, this will throw an exn bc coo would need .get(), right? The if condition needs to be just on the engine name+version, right?